editor/shader_create_dialog.cpp

```diff
@@ -371,9 +372,6 @@ String ShaderCreateDialog::_validate_path(const String &p_path, bool p_file_must
 	if (f->dir_exists(p)) {
 		memdelete(f);
 		return TTR("A directory with the same name exists.");
-	} else if (p_file_must_exist && !f->file_exists(p)) {
-		memdelete(f);
```

Comment:
You can simplify and remove the need for the `memdelete(f)` and `memdelete(d)` by using `FileAccessRef` and `DirAccessRef` instead of `FileAccess *` and `DirAccess *`.

Reply:
Done